### PR TITLE
feat: Auto detect light & dark theme (ACC-304)

### PR DESCRIPTION
**Pull request by thisisamir98**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/ACC-304" title="ACC-304" target="_blank">StoryACC-304</a>  Screen Reader announcement for Typing Indicator
  </td></table>
  <br />
 

Solution until we fully implement Auto/light/dark in preferences

### CODE REVIEW

File: src/script/hooks/useAutomaticTheme.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+export const THEMES = {
+  DARK: 'dark',
+  DEFAULT: 'default',
+};
+
+export const useSystemTheme = () => {
+  const isSystemDarkMode = useMatchMedia('(prefers-color-scheme: dark)');
+  return isSystemDarkMode;
+};
```
Comment (przemvs) [inline src/script/hooks/useAutomaticTheme.ts]: We need this hook? It's used only inside useAutomaticHook.

File: src/script/hooks/useAutomaticTheme.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    amplify.subscribe(WebAppEvents.PROPERTIES.UPDATE.INTERFACE.THEME, setTheme);
+    amplify.subscribe(WebAppEvents.PROPERTIES.UPDATED, (properties: WebappProperties) =>
+      setTheme(properties.settings.interface.theme),
+    );
+  }, []);
+
+  const isSystemDarkMode = useSystemTheme();
+
+  useEffect(() => {
+    const nextTheme = (isSystemDarkMode ? THEMES.DARK : THEMES.DEFAULT) as ThemeType;
```
Comment (przemvs) [inline src/script/hooks/useAutomaticTheme.ts:66]:
```suggestion
    const nextTheme: ThemeType = (isSystemDarkMode ? THEMES.DARK : THEMES.DEFAULT);
```

This one doesn';t work ? 🤔

File: src/script/hooks/useAutomaticTheme.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+import {useMatchMedia} from '@wireapp/react-ui-kit';
+import {WebAppEvents} from '@wireapp/webapp-events';
+
+import {PropertiesRepository} from '../properties/PropertiesRepository';
+import {PROPERTIES_TYPE} from '../properties/PropertiesType';
+
+type ThemeType = 'dark' | 'default';
+
+export const updateTheme = (newTheme: ThemeType) => {
```
Comment (przemvs) [inline src/script/hooks/useAutomaticTheme.ts]: We need to export this updateTheme ? 🤔

File: src/script/hooks/useAutomaticTheme.ts (hunk 2 above)
Comment (V-Gira) [reply src/script/hooks/useAutomaticTheme.ts:66]: Also, doesn't this line force the user to use system settings, leaving them stuck on light mode if the system doesn't define a preference?

Comment (thisisamir98) [reply src/script/hooks/useAutomaticTheme.ts:66]: image

File: src/script/hooks/useAutomaticTheme.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+const THEMES_CLASS_PREFIX = 'theme-';
+
+export const THEMES = {
+  DARK: 'dark',
+  DEFAULT: 'default',
+};
+
+const useAutomaticTheme = (propertiesRepository: PropertiesRepository): ThemeType => {
+  const [theme, setTheme] = useState<ThemeType>(propertiesRepository.getPreference(PROPERTIES_TYPE.INTERFACE.THEME));
```
Comment (przemvs) [inline src/script/hooks/useAutomaticTheme.ts:49]: propertiesRepository.getPreference(PROPERTIES_TYPE.INTERFACE.THEME) - move this to variable, you have duplicated it :)

**Review by V-Gira — COMMENTED** [review]
I think we should talk with design and insist to implement the radio instead of checkbox now. The way it ignores your user settings every time you start the app is going to be annoying fast

**Pull request closed without merging**